tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -380,46 +400,55 @@ def test_poly1d_ne2(self, xp, dtype):
 
 @testing.gpu
 @testing.parameterize(*testing.product({
-    'fname': ['polyadd', 'polysub'],
+    'fname': ['polyadd', 'polysub', 'polymul'],
     'shape1': [(), (0,), (3,), (5,)],
-    'shape2': [(), (0,), (3,), (5,)]
+    'shape2': [(), (0,), (3,), (5,)],
 }))
 class TestPolyArithmeticShapeCombination(unittest.TestCase):
 
     @testing.for_all_dtypes(no_bool=True)
-    @testing.numpy_cupy_array_equal()
+    @testing.numpy_cupy_allclose(rtol=1e-5)
     def test_polyroutine(self, xp, dtype):
         func = getattr(xp, self.fname)
         a = testing.shaped_arange(self.shape1, xp, dtype)
         b = testing.shaped_arange(self.shape2, xp, dtype)
-        with cupyx.allow_synchronize(False):
-            return func(a, b)
+        return func(a, b)
 
 
 @testing.gpu
 @testing.parameterize(*testing.product({
-    'fname': ['polyadd', 'polysub'],
+    'fname': ['polyadd', 'polysub', 'polymul'],
 }))
 class TestPolyArithmeticDiffTypes(unittest.TestCase):
 
     @testing.for_all_dtypes_combination(names=['dtype1', 'dtype2'])
-    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    @testing.numpy_cupy_allclose(rtol=1e-5, accept_error=TypeError)
     def test_polyroutine_diff_types_array(self, xp, dtype1, dtype2):
         func = getattr(xp, self.fname)
         a = testing.shaped_arange((10,), xp, dtype1)
         b = testing.shaped_arange((5,), xp, dtype2)
-        with cupyx.allow_synchronize(False):
-            return func(a, b)
+        return func(a, b)
 
     @testing.for_all_dtypes_combination(names=['dtype1', 'dtype2'])
-    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    @testing.numpy_cupy_allclose(rtol=1e-5, accept_error=TypeError)
     def test_polyroutine_diff_types_poly1d(self, xp, dtype1, dtype2):
         func = getattr(xp, self.fname)
         a = testing.shaped_arange((10,), xp, dtype1)
         b = testing.shaped_arange((5,), xp, dtype2)
         a = xp.poly1d(a, variable='z')
         b = xp.poly1d(b, variable='y')
-        with cupyx.allow_synchronize(False):
-            out = func(a, b)
+        out = func(a, b)
         assert out.variable == 'x'
         return out
+
+
+@testing.gpu
+class TestPolymul(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    def test_polymul_ndim(self, dtype):
```

Comment:
Why not check it also for `polyadd` and `polysub`?

Reply:
https://github.com/cupy/cupy/pull/3548#discussion_r454143443

Reply:
I see, but does `numpy.polymul` guarantee to raise an error for such inputs?

Reply:
Yes because `numpy.polymul` calls `poly1d().coeffs` on its inputs first, wich raises `ValueError` for multidimensional inputs.

Reply:
Is it documented?

Reply:
It is not stated that it raises error for ndim inputs. But it says that the input must be poly1d object or 1D sequence of polynomial coefficients

Reply:
The documentation of `numpy.polyadd` also says that `Each input must be either a poly1d object or a 1D sequence of polynomial coefficients` (https://numpy.org/doc/stable/reference/generated/numpy.polyadd.html), but `numpy.polyadd` does not always raise an error for non-1dim inputs.

Reply:
`polyadd` raises error for most of the cases. Only few cases (technically speaking where broadcasting is possible), it does not. The same applies for polysub. It is not clearly stated in any of the 3 documentations that it raises error for these cases. But polymul's implementation in numpy does guarantee raising error for ndim inputs. So what do you suggest now? Removing the test or keep it?

Reply:
How they are implemented internally does not become the reason for the difference in the behavior of `numpy.polyadd` and `numpy.polymul`.

Reply:
I think it's both OK to keep the test or remove it. However, if you want to keep the test, please fix all routines to raise `ValueError` for all non-1d inputs.